.github/workflows/ci.yaml

```
@@ -0,0 +1,31 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  tests:
+    name: "Tests on Python ${{ matrix.python-version }} for ${{ matrix.runs-on }}"
+    runs-on: "${{ matrix.runs-on }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.12"]
+        runs-on: ["ubuntu-latest", "macos-latest", "windows-latest"]
```

I think we can have a smaller size of matrix.
```suggestion
        runs-on: ["ubuntu-latest", "macos-latest"]
```